### PR TITLE
[bug] fix primary entity identification bug

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR aims to fix primary entity identification bug for the time-aggregated-item-aggregated feature. Below is the example of such feature:
```
# Generated by SDK version: 0.3.0.dev10
from bson import ObjectId
from featurebyte import EventTable
from featurebyte import FeatureJobSetting
from featurebyte import ItemTable


# item_table name: "LABOBSERVATION", event_table name: "LABRESULT"
item_table = ItemTable.get_by_id(ObjectId("64874aa0f48964db9d3e9847"))
item_view = item_table.get_view(
    event_suffix=None,
    view_mode="manual",
    drop_column_names=[],
    column_cleaning_operations=[],
    event_drop_column_names=["record_available_at"],
    event_column_cleaning_operations=[],
    event_join_column_names=["ReportDate", "LabResultGuid", "PatientGuid"],
)
col = item_view["IsAbnormalValue"]
view = item_view.copy()
view["IsAbnormalValueInt"] = (col == "true").astype(int)
col_1 = view.groupby(by_keys=["LabResultGuid"], category=None).aggregate(
    value_column="IsAbnormalValueInt",
    method="sum",
    feature_name="Abnormal Values count by labresult",
    skip_fill_na=True,
)
col_2 = col_1.copy()
col_2[col_1.isnull()] = 0
col_2.name = "Abnormal Values count by labresult"

# event_table name: "LABRESULT"
event_table = EventTable.get_by_id(ObjectId("64874a9af48964db9d3e9846"))
event_view = event_table.get_view(
    view_mode="manual",
    drop_column_names=["record_available_at"],
    column_cleaning_operations=[],
)
joined_view = event_view.add_feature(
    new_column_name="number_of_abnormal_values",
    feature=col_2,
    entity_column="LabResultGuid",
)
grouped = joined_view.groupby(
    by_keys=["PatientGuid"], category=None
).aggregate_over(
    value_column="number_of_abnormal_values",
    method="sum",
    windows=["90d"],
    feature_names=["Number of abnormal results in the Last 90d"],
    feature_job_setting=FeatureJobSetting(
        blind_spot="64800s", frequency="86400s", time_modulo_frequency="5s"
    ),
    skip_fill_na=True,
)
feat = grouped["Number of abnormal results in the Last 90d"]
feat_1 = feat.copy()
feat_1[feat.isnull()] = 0
feat_1.name = "Number of abnormal results in the Last 90d"
output = feat_1
output.save(_id=ObjectId("64874c6f1e2c4b4eeb6e6fe0"))
```

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/transform/entity_extractor.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        self,
+        branch_state: EntityExtractorBranchState,
+        global_state: EntityExtractorGlobalState,
+        node: Node,
+        input_node_names: List[str],
+    ) -> Tuple[List[str], bool]:
+        skip_input_nodes = False
+        if isinstance(node.parameters, BaseGroupbyParameters):
+            # if groupby node has entity_ids, skip further exploration on input nodes
+            if node.parameters.entity_ids:
```
Comment (YS-L) [inline featurebyte/query_graph/transform/entity_extractor.py:44]: It is also possible to use an empty list as the groupby keys to create purely time based features without any entity. Will that be handled correctly by this condition?

Comment (gee-senbong) [reply featurebyte/query_graph/transform/entity_extractor.py:44]: If it is possible to create such a feature, we should move this `skip_input_nodes` outside. Will update the logic here.

Comment (gee-senbong) [reply featurebyte/query_graph/transform/entity_extractor.py:44]: Updated the logic here & added test to cover this case.

**Review by YS-L — APPROVED** [review]
LGTM

**Pull request merged**